### PR TITLE
Use the entire podTemplateSpec to spawn the realm-import job

### LINKED ISSUE #11847: Operator - Job needs imagePullSecret support
### Describe the bug

I was able to create the Keycloak deployment instance with imagePullSecret using unsupported podTemplate.
Unfortunately, when the operator create a realm import job it doesn't take the unsupported podTemplate and the image cannot be pulled. 

### Version

18.0.0

### Expected behavior

Use the unsupported podTemplate or provide a way to specify the imagePullSecret to use. 

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

I will try to add the imagePullSecret to the serviceaccount as a workaround in waiting for a solution.

### PR DESCRIPTION
**Pull request by andreaTP**
Resolves #11847

### CODE REVIEW

**Review by pedroigor — DISMISSED** [review]

**Comment by andreaTP** [comment]
Test failures are (probably) legit, will look into tomorrow.

**Comment by andreaTP** [comment]
Ci is green now 🎉

File: operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java [hunk 1]
```diff
@@ -118,6 +118,8 @@ public void testKeycloakCacheIsConnected() throws Exception {
                     Log.info("Checking Keycloak is stable.");
                     CRAssert.assertKeycloakStatusCondition(crSelector.get(), KeycloakStatusCondition.READY, true);
                 });
+        // Remove the completed pod for the job
+        realmImportSelector.delete();
```
Comment (vmuzikar) [inline operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java:122]: Out of curiosity, why is this change necessary?

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks @andreaTP!

File: operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java (hunk 1 above)
Comment (andreaTP) [reply operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java:122]: Good question!

 - By applying the entire template the Pod of the Job is inheriting more labels
 - we are (a little below in this test) testing the connection toward each single Pods selected by labels
 - now the Job Pod is included in the selection but it's in "Completed" state (e.g. not running)
 
 Solution:
  - removing the realm import CR deletes all the connected resources (and the Pod too)

Comment (vmuzikar) [reply operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java:122]: Thank you for clarifying!

**Review by pedroigor — APPROVED** [review]

**Pull request merged**